reckless_github/src/repository.rs

```diff
@@ -43,82 +50,47 @@ impl Github {
-                                &path_to_plugin,
-                                PluginLang::Python,
-                            ));
-                        }
-                        "go.mod" => {
-                            self.plugins.push(Plugin::new(
-                                &plugin_name,
-                                &path_to_plugin,
-                                PluginLang::Go,
-                            ));
-                        }
-                        "cargo.toml" => {
-                            self.plugins.push(Plugin::new(
-                                &plugin_name,
-                                &path_to_plugin,
-                                PluginLang::Rust,
-                            ));
-                        }
-                        "pubspec.yaml" => {
-                            self.plugins.push(Plugin::new(
-                                &plugin_name,
-                                &path_to_plugin,
-                                PluginLang::Dart,
-                            ));
-                        }
-                        "package.json" => {
-                            self.plugins.push(Plugin::new(
-                                &plugin_name,
-                                &path_to_plugin,
-                                PluginLang::JavaScript,
-                            ));
-                        }
-                        "tsconfig.json" => {
-                            // FIXME: avoid to unwrap here
-                            self.plugins.push(Plugin::new(
-                                plugin_name.as_str(),
-                                path_to_plugin.as_str(),
-                                PluginLang::TypeScript,
-                            ));
-                        }
-                        "reckless.yml" | "reckless.yaml" => {
-                            let conf_path = format!("{}/{}", path_to_plugin, file_name);
-                            let mut conf_str = String::new();
-                            File::open(conf_path.as_str())
-                                .await?
-                                .read_to_string(&mut conf_str)
-                                .await?;
-                            let _: Conf = serde_yaml::from_str(&conf_str).unwrap();
-                            // FIXME: store the conf inside the plugin
-                        }
-                        _ => continue,
+                            "reckless.yml" | "reckless.yaml" => {
+                                let conf_path = format!("{}/{}", path_to_plugin, file_name);
+                                let mut conf_str = String::new();
+                                File::open(conf_path.as_str())
+                                    .await?
+                                    .read_to_string(&mut conf_str)
+                                    .await?;
+                                conf = serde_yaml::from_str(&conf_str).unwrap();
```

Comment:
The conf is decoded in the correct Configuration type?

In addition, can you store the configuration somewhere in the plugin so we can make the PR ready? otherwise this is still not completed

Reply:
It would seem so.
Also I have stored it [here](https://github.com/cln-reckless/reckless.rs/blob/8bd7a90990f1ec9aac69e39a699b870f2f072c96/reckless_lib/src/plugin.rs#L85)
assigned [here](https://github.com/cln-reckless/reckless.rs/blob/8bd7a90990f1ec9aac69e39a699b870f2f072c96/reckless_lib/src/plugin.rs#L95)

Reply:
I do not think that this will work, there is no type in there! but maybe you did not test the code, so for the next time please test it